bbq/vm/vm.go

```diff
@@ -813,6 +815,21 @@ func (vm *VM) run() {
 	}
 }
 
+func onEmitEvent(vm *VM, ins opcode.InstructionEmitEvent) {
+	eventType := vm.loadType(ins.TypeIndex).(*interpreter.CompositeStaticType)
```

Comment:
Maybe move this to after the if condition `if onEventEmitted == nil { }`. i.e: We wouldn't need to load the type if the callback is not set.

Reply:
Good idea! Done in 9ee137fd6056cdf0edbc7e53094f16a08f8073ba